src/intelligence_layer/evaluation/dataset/studio_dataset_repository.py

```diff
@@ -0,0 +1,195 @@
+        """Returns a dataset identified by the given dataset ID.
+
+        Args:
+            dataset_id: Dataset ID of the dataset to delete.
+
+        Returns:
+            :class:`Dataset` if it was not, `None` otherwise.
+            :exception: Raises DataInternalError exception when the client fails.
+        """
+        remote_dataset = self.data_client.get_dataset(
+            repository_id=self.repository_id, dataset_id=dataset_id
+        )
+        return Dataset(
+            id=remote_dataset.dataset_id,
+            name=remote_dataset.name or "",
+            labels=set(remote_dataset.labels)
+            if remote_dataset.labels is not None
+            else set(),
+            metadata=remote_dataset.metadata or dict(),
+        )
+
+    def datasets(self) -> Iterable[Dataset]:
+        """Returns all :class:`Dataset`s sorted by their ID.
+
+        Returns:
+            :class:`Sequence` of :class:`Dataset`s.
+            :exception: Raises DataInternalError exception when the client fails.
+        """
+        for remote_dataset in self.data_client.list_datasets(
+            repository_id=self.repository_id
+        ):
+            yield Dataset(
+                id=remote_dataset.dataset_id,
+                name=remote_dataset.name or "",
+                labels=set(remote_dataset.labels)
+                if remote_dataset.labels is not None
+                else set(),
+                metadata=remote_dataset.metadata or dict(),
+            )
+
+    def dataset_ids(self) -> Iterable[str]:
+        """Returns all sorted dataset IDs.
+
+        Returns:
+            :class:`Iterable` of dataset IDs.
+            :exception: Raises DataInternalError exception when the client fails.
+        """
+        datasets = self.data_client.list_datasets(repository_id=self.repository_id)
+        return (dataset.dataset_id for dataset in datasets)
+
+    def example(
+        self,
+        dataset_id: str,
+        example_id: str,
+        input_type: type[Input],
+        expected_output_type: type[ExpectedOutput],
+    ) -> Optional[Example[Input, ExpectedOutput]]:
+        """Returns an :class:`Example` for the given dataset ID and example ID.
+
+        Args:
+            dataset_id: Dataset ID of the linked dataset.
+            example_id: ID of the example to retrieve.
+            input_type: Input type of the example.
+            expected_output_type: Expected output type of the example.
+
+        Returns:
+            :class:`Example` if it was found, `None` otherwise.
+            :exception: Raises DataInternalError exception when the client fails.
+        """
+        stream = self.data_client.stream_dataset(
+            repository_id=self.repository_id, dataset_id=dataset_id
+        )
+        for item in stream:
+            data = json.loads(item.decode())
+            if data["id"] == example_id:
+                return Example[input_type, expected_output_type].model_validate(data)  # type: ignore
```

Comment:
This operation is quite expensive for singular item access. Is there a more efficient way to do it?

Reply:
and since the `examples` function uses yield, you might as well use that here instead of streaming it again.

Reply:
As transforming str -> json -> pydantic is quite expensive(python everything is an object), I prefer to call the stream(fast enough) function again. About of the `expensive` operation this is what we are doing in [file](https://github.com/Aleph-Alpha/intelligence-layer-sdk/blob/d2d53a08aadc5b250255314a6687ce9255201400/src/intelligence_layer/evaluation/dataset/file_dataset_repository.py#L100)  and [memory](https://github.com/Aleph-Alpha/intelligence-layer-sdk/blob/d2d53a08aadc5b250255314a6687ce9255201400/src/intelligence_layer/evaluation/dataset/in_memory_dataset_repository.py#L72) repositories, I try to follow the same for now.

Reply:
The overhead we would have would be json -> pydantic only, which is probably nothing compared to the cost of network traffic, but you are right. For the file and memory repositories, fetching the dataset isn't as expensive as here with network, so I believe we didn't bother with making faster. Maybe its also not a problem here

Reply:
As streaming over HTTP can reach even ~10-12MB/s, I think it's okay for the current use case. What do you think?

Reply:
probably ok for now, we can revisit it once its an issue